### PR TITLE
[Documentation] Update DataProviders.md

### PR DESCRIPTION
**Pull request by benwinding**
Added an npm package for connecting to the **Firebase - Firestore** database, it's kept up to date and being used by many clients already with great success.

### CODE REVIEW

File: docs/DataProviders.md [hunk 1]
```diff
@@ -72,7 +72,10 @@ Due to the breaking changes, the following providers are no longer working with
 
 * **[DynamoDb](https://github.com/abiglobalhealth/aor-dynamodb-client)**: [abiglobalhealth/aor-dynamodb-client](https://github.com/abiglobalhealth/aor-dynamodb-client)
 * **[Epilogue](https://github.com/dchester/epilogue)**: [dunghuynh/aor-epilogue-client](https://github.com/dunghuynh/aor-epilogue-client)
-* **[Loopback](http://loopback.io/)**: [kimkha/aor-loopback](https://github.com/kimkha/aor-loopback)
```
Comment (djhi) [inline docs/DataProviders.md]: Why did you remove this one ?

**Comment by benwinding** [comment]
Whoops sorry about that, I was just trying to fix the first commit, not intentional.

File: docs/DataProviders.md (hunk 1 above)
Comment (benwinding) [reply docs/DataProviders.md]: Sorry that was a mistake, I was trying to fix the first commit to allow for an automatic merge

File: docs/DataProviders.md [hunk 2]
```diff
@@ -72,7 +72,10 @@ Due to the breaking changes, the following providers are no longer working with
...
+* **[Firebase](https://firebase.google.com/)**: [sidferreira/aor-firebase-client](https://github.com/sidferreira/aor-firebase-client)		
+* **[Firebase - Firestore](https://firebase.google.com/docs/firestore)**: [benwinding/react-admin-firebase](https://github.com/benwinding/react-admin-firebase)
```
Comment (fzaninotto) [inline docs/DataProviders.md]: I don't understand: why did you add that react-admin provider to the list of legacy admin-on-rest providers?

File: docs/DataProviders.md [hunk 3]
```diff
@@ -72,7 +72,10 @@ Due to the breaking changes, the following providers are no longer working with
 
 * **[DynamoDb](https://github.com/abiglobalhealth/aor-dynamodb-client)**: [abiglobalhealth/aor-dynamodb-client](https://github.com/abiglobalhealth/aor-dynamodb-client)
 * **[Epilogue](https://github.com/dchester/epilogue)**: [dunghuynh/aor-epilogue-client](https://github.com/dunghuynh/aor-epilogue-client)
-* **[Loopback](http://loopback.io/)**: [kimkha/aor-loopback](https://github.com/kimkha/aor-loopback)
+* **[Firebase](https://firebase.google.com/)**: [sidferreira/aor-firebase-client](https://github.com/sidferreira/aor-firebase-client)
```
Comment (fzaninotto) [inline docs/DataProviders.md]: I'm -1 with listing admin-on-rest packages here that already have a react-admin version. It only adds noise.

File: docs/DataProviders.md (hunk 2 above)
Comment (djhi) [reply docs/DataProviders.md]: Oups, didn't notice...
